load.php

```diff
@@ -260,3 +270,53 @@ function perflab_load_active_and_valid_modules() {
 if ( is_admin() ) {
 	require_once PERFLAB_PLUGIN_DIR_PATH . 'admin/load.php';
 }
+
+/**
+ * Trigger actions when a module gets activated or deactivated.
+ *
+ * @since n.e.x.t
+ *
+ * @param string $option    The option name.
+ * @param mixed  $old_value Old value of the option.
+ * @param mixed  $value     New value of the option.
+ *
+ * @return void
+ */
+function perflab_run_module_activation_deactivation( $option, $old_value, $value ) {
+	if ( PERFLAB_MODULES_SETTING !== $option ) {
+		return;
+	}
+
+	// Get the list of modules that were activated, and load the activate.php files if they exist.
+	foreach ( $value as $module => $module_settings ) {
+		if ( ! empty( $module_settings['enabled'] ) && ( empty( $old_value[ $module ] ) || empty( $old_value[ $module ]['enabled'] ) ) ) {
+			$module_activation_file = PERFLAB_PLUGIN_DIR_PATH . 'modules/' . $module . '/activate.php';
+			if ( ! file_exists( $module_activation_file ) ) {
+				continue;
+			}
+			$module = require $module_activation_file;
+			if ( ! is_callable( $module ) ) {
+				continue;
+			}
+			$module();
+		}
+	}
+
+	// Get the list of modules that were deactivated, and load the deactivate.php files if they exist.
+	foreach ( $old_value as $module => $module_settings ) {
+		if ( ! empty( $module_settings['enabled'] ) && ( empty( $value[ $module ] ) || empty( $value[ $module ]['enabled'] ) ) ) {
+			$module_deactivation_file = PERFLAB_PLUGIN_DIR_PATH . 'modules/' . $module . '/deactivate.php';
+			if ( ! file_exists( $module_deactivation_file ) ) {
+				continue;
+			}
+			$module = require $module_deactivation_file;
+			if ( ! is_callable( $module ) ) {
+				continue;
+			}
+			$module();
+		}
+	}
+
+	return $value;
+}
+add_action( 'update_option', 'perflab_run_module_activation_deactivation', 10, 3 );
```

Comment:
In addition to the `update_option*` action, we also need to trigger this on the related `add_option*` action (e.g. `'add_option_' . PERFLAB_MODULES_SETTING`).
We can use e.g. a closure to fix the parameter there; basically instead of `$old_value` (which isn't present in the `add_option*` actions) we can pass the default value to the method, this way the callback logic can remain the same for both actions.

Reply:
Done